doc/kci_testsuite.md

```diff
@@ -0,0 +1,300 @@
+
+```
+chmod +x jenkins/debian/debos/overlays/btrfs/usr/bin/btrfs-verify-cli.sh
+chmod +x jenkins/debian/debos/script/buster-btrfs.sh
+```
+
+6. Now build `buster-btrfs` rootfs image using kci_rootfs. 
+
+7. Once rootfs image is created, you can use it with appropriate kernel image. Make sure that kernel config has 
+   appropriate btrfs configs and other file system specific entries enabled. 
+
+8. Add appropriate test entry with LAVA job definition for `btrfs-verify-cli.sh`. For example,
+
+```
+run:
+    steps:
+        - lava-test-case test-btrfs-cli-001 --shell ls /usr/bin/btrfs-verify-cli.sh
+
+```
+
+If you like read more about LAVA job definition, please check [here](https://docs.lavasoftware.org/lava/writing-tests.html)
+
+
+### How to add tests to LAVA
+
+#### Prerequisites
+
+- Rootfs image is built 
+- Rootfs image boots on a platform
+- Tests can be run
+
+If you're facing issues with tha above you may want to go through the section "How to add test suite to a rootfs image"
+
+#### Adding new test plan to LAVA
+
+1. Use the rootfs image to boot the device and produce a LAVA job definition for the tests.
+Upload rootfs image to kernelci-backend (storage)
+
+```
+kci_rootfs upload --rootfs-dir /path/to/rootf/image/ --upload-path server/path --api http://api.kernelci --token KCI_API_TOKEN
+```
+
+Add rootfs definition to `test-configs.yaml`
+
+```yaml
+  debian_buster-v4l2_ramdisk:
+    type: debian
+    ramdisk: 'buster-v4l2/20200626.0/{arch}/rootfs.cpio.gz'
+```
+
+`ramdisk` path must be relative to storage server URL.
+
+Detailed description of the test configuration can be found [here](https://github.com/kernelci/kernelci-doc/wiki/Test-configurations)
+
+2. Add LAVA job definition template
+
+Create appropriate directory structure under `templates` directory.
+
+```
+mkdir templates/v4l2-compliance/
+```
+
+Create test plan definition
+
+```
+cat templates/v4l2-compliance/v4l2-compliance.jinja2
+```
+
+```yaml
+- test:
+    timeout:
+      minutes: 10
+    definitions:
+    - repository:
+        metadata:
+          format: Lava-Test Test Definition 1.0
+          name: {{ plan }}
+          description: "v4l2 test plan"
+          os:
+          - debian
+          scope:
+          - functional
+        run:
+          steps:
+          - /usr/bin/v4l2-parser.sh {{ v4l2_driver }}
+      from: inline
+      name: {{ plan }}
+      path: inline/{{ plan }}.yaml
+```
+
+
+```
+cat templates/v4l2-compliance/generic-qemu-v4l2-compliance-template.jinja2 
+```
+
+```
+{% extends 'boot/generic-qemu-boot-template.jinja2' %}
+{% block actions %}
+{{ super () }}
+
+{% include 'v4l2-compliance/v4l2-compliance.jinja2' %}
+
+{% endblock %}
+
+{%- block image_arg %}
+        image_arg: '-kernel {kernel} -append "console={{ console_dev }},115200 root=/dev/ram0 debug verbose {{ extra_kernel_args }}"'
+{%- endblock %}
+```
+
+3. Update `test-plans` section of `test-configs.yaml` with new test definition
+
+
+```yaml
```

Comment:
This one is a bit advanced because it's a variant with some parameters.  Maybe a simpler one which wouldn't need `pattern` and `params` would be better first to cover the regular cases, then showing an advanced example with details about extra options would be useful as a second step.